massa-ledger-worker/src/ledger_db.rs

```diff
@@ -284,6 +284,9 @@ impl LedgerDB {
     /// * `ledger_entry`: complete entry to be added
     /// * `batch`: the given operation batch to update
     fn put_entry(&self, addr: &Address, ledger_entry: LedgerEntry, batch: &mut DBBatch) {
+        // Ensures any potential previous entry is fully deleted.
+        self.delete_entry(addr, batch);
+
         let db = self.db.read();
```

Comment:
Calling `self.delete_entry(addr, batch);` directly duplicates many operations (db.read(), various serialization).

Maybe calling `db.delete_key(batch, serialized_key);` right before each `db.put_or_update_entry_value(batch, serialized_key...` is a bit more efficient.